src/github.com/couchbase/sync_gateway/rest/server_context.go

```diff
@@ -632,3 +650,30 @@ func collectAccessRelatedWarnings(config *DbConfig, context *db.DatabaseContext)
 	return []string{}
 
 }
+
+func takeDbOffline(dbctx *db.DatabaseContext) error {
```

Comment:
This method looks like it will panic if dbctx comes in nil - should have a check for that.